src/applications/strings/strings.cc

```diff
@@ -210,3 +210,79 @@ bool mata::applications::strings::is_lang_eps(const Nfa& aut) {
     }
     return true;
 }
+
+std::optional<std::vector<mata::Word>> mata::applications::strings::get_words_of_lengths(const Nft& nft, std::vector<unsigned> lengths) {
+    assert(nft.num_of_levels == lengths.size());
+    assert(!nft.contains_jump_transitions());
+    if (nft.initial.empty() || nft.final.empty()) { return std::nullopt; }
+    if (nft.initial.intersects_with(nft.final) && std::all_of(lengths.begin(), lengths.end(), [](int x) { return x == 0; })) { return std::vector<mata::Word>(nft.num_of_levels, mata::Word()); }
+
+    std::vector<mata::Word> result(lengths.size());
+    for (const State initial_state: nft.initial) {
+        /// Current state, its state post iterator, its end iterator, and iterator in the current symbol post to target states.
+        std::vector<std::tuple<State, StatePost::const_iterator, StatePost::const_iterator, StateSet::const_iterator>> worklist{};
+        std::vector<mata::Word> result(lengths.size());
+        auto is_result_correct = [&result, &lengths]() {
```

Comment:
```suggestion
        auto is_result_correct = [&]() {
```
Note: Only `[&]` is enough. No need to explicitly enumerate all shared variables.

Reply:
I like to explicitly write everything I want to use, so that I do not accidentally use something that was not intended to be used.